cubi_tk/sodar/ingest_fastq.py

```diff
@@ -81,60 +82,203 @@ def setup_argparse(cls, parser: argparse.ArgumentParser) -> None:
+                )
+                res = 1
+
+        return res
+
+    def get_project_uuid(self, lz_uuid: str):
+        """Get project UUID from landing zone UUID.
+        :param lz_uuid: Landing zone UUID.
+        :type lz_uuid: str
+
+        :return: Returns Sodar UUID of corresponding project.
+        """
+        from sodar_cli.api import landingzone
+
+        lz = landingzone.retrieve(
+            sodar_url=self.args.sodar_url,
+            sodar_api_token=self.args.sodar_api_token,
+            landingzone_uuid=lz_uuid,
+        )
+        return lz.project
 
     def build_base_dir_glob_pattern(self, library_name: str) -> typing.Tuple[str, str]:
         raise NotImplementedError(
             "build_base_dir_glob_pattern() not implemented in SodarIngestFastq!"
         )
 
+    def get_match_to_collection_mapping(
+        self, project_uuid: str, in_column: str, out_column: typing.Optional[str] = None
+    ) -> typing.Dict[str, str]:
+        """Return a dict that matches all values from a specific `ìn_column` of the assay table
+        to a corresponding `out_column` (default if not defined: last Material column)."""
+
+        # This part is only needed to get `assay.file_name`
+        # -> could be removed if we can get around that
+        investigation = api.samplesheet.retrieve(
+            sodar_url=self.args.sodar_url,
+            sodar_api_token=self.args.sodar_api_token,
+            project_uuid=project_uuid,
+        )
+        assay = None
+        for study in investigation.studies.values():
+            for assay_uuid in study.assays.keys():
+                if (self.args.assay is None) and (assay is None):
+                    assay = study.assays[assay_uuid]
+                if (self.args.assay is not None) and (self.args.assay == assay_uuid):
+                    assay = study.assays[assay_uuid]
+                    logger.info("Using irods path of assay %s: %s", assay_uuid, assay.irods_path)
+                    break
+
+        isa_dict = api.samplesheet.export(
+            sodar_url=self.args.sodar_url,
+            sodar_api_token=self.args.sodar_api_token,
+            project_uuid=project_uuid,
+        )
+
+        assay_tsv = isa_dict["assays"][assay.file_name]["tsv"]
+        assay_header, *assay_lines = assay_tsv.rstrip("\n").split("\n")
+        assay_header = assay_header.split("\t")
+        assay_lines = map(lambda x: x.split("\t"), assay_lines)
+
+        # Never match these assay cols
+        ignore_cols = (
+            "Performer",
+            "Date",
+            "Protocol REF",
+            "Unit",
+            "Term Source REF",
+            "Term Accession Number",
+        )
+
+        in_column_index = [
+            i
+            for i, head in enumerate(assay_header)
+            if head not in ignore_cols
+            and in_column.lower()
+            in re.sub("(Parameter Value|Comment|Characteristics)", "", head).lower()
```

Comment:
Sort of, I want to remove these part of the column names since they are "invisible" and if people have a column named i.e. 'Comment' then checking for that string as part of the name will make problems. The brackets should not have that issue and removing both them requires a much more complicated regex.